crates/core/tedge/src/cli/connect/jwt_token.rs

```diff
@@ -7,13 +7,14 @@ use rumqttc::MqttOptions;
 use rumqttc::Outgoing;
 use rumqttc::Packet;
 use rumqttc::QoS::AtLeastOnce;
+use tedge_config::IpAddress;
 
-pub(crate) fn get_connected_c8y_url(port: u16, host: String) -> Result<String, ConnectError> {
+pub(crate) fn get_connected_c8y_url(port: u16, host: IpAddress) -> Result<String, ConnectError> {
```

Comment:
As above, I think that may be incorrect because hosts can be also DNS names, but I'm not sure

Reply:
@Bravo555 is correct we need to be able to give a host name.

Reply:
If that's the case, the logic calling this that can only ever be an `IpAddress` is broken. I presume it's a case of us wanting to use the MQTT client address rather than the bind address? My gut feeling there is that the client address default should be the bind address (whereas the current default is just a copy of the bind address default)?